### PR TITLE
GH-1494: As a user I want support for optional chaining (aka elvis operator, etc.) (part 2)

### PR DESCRIPTION
**Pull request by mor-n4**
See #1494.

This part includes the transpilation of nullish coalescing operator and optional chaining to ES6.

### CODE REVIEW

**Comment by mor-n4** [comment]
@mmews-n4 Please double check my assumption that a single helper variable will always be sufficient, not matter how many `?.` and `??` are used and no matter how they are nested.

**Comment by mor-n4** [comment]
Thanks! Builds green, merging ...

File: plugins/org.eclipse.n4js.transpiler.es/src/org/eclipse/n4js/transpiler/es/transform/ExpressionTransformation.xtend [hunk 1]
```diff
@@ -219,11 +238,106 @@ class ExpressionTransformation extends Transformation {
...
+		} else {
+			replace(coalExpr, replacement);
+		}
+	}
+
+	/**
+	 * Converts
+	 * <pre>a?.b.c</pre>
+	 * to
+	 * <pre>($temp = a) == null ? void 0 : $temp.b.c</pre>
```
Comment (szarnekow) [inline plugins/org.eclipse.n4js.transpiler.es/src/org/eclipse/n4js/transpiler/es/transform/ExpressionTransformation.xtend]: Is the `==null` check rewritten in subsequent step?

Comment (mor-n4) [reply plugins/org.eclipse.n4js.transpiler.es/src/org/eclipse/n4js/transpiler/es/transform/ExpressionTransformation.xtend]: If the missing check for undefined is what you mean:
the cool kidz use `x == null` for `x === void 0 || x === null` these days (as I was told), they're equivalent (I know ... it's Javascript 😄 ).

Comment (szarnekow) [reply plugins/org.eclipse.n4js.transpiler.es/src/org/eclipse/n4js/transpiler/es/transform/ExpressionTransformation.xtend]: Yes, that I was thinking about. Thank you for enlightening me, too :)

**Pull request merged**